components/builder-depot/src/backend/s3.rs

```diff
@@ -103,6 +102,25 @@ impl S3Handler {
         }
     }
 
+    // This function checks whether an uploaded file
+    // exists in the configured s3 bucket. It should
+    // only get called from within an upload future.
+    fn object_exists(&self, object_key: &str) -> Result<()> {
+        let mut request = ListObjectsRequest::default();
+        request.bucket = self.bucket.clone();
+        request.prefix = Some(object_key.to_string());
+
+        match self.client.list_objects(&request).sync() {
+            Ok(object_list) => match object_list {
+                object_key => {
```

Comment:
This match seems a bit redundant - is it needed (does not seem like there is a case for object key not in the list)..?